### ISSUE TITLE
genvalidity-hspec-0.6.1.0 build failure with hspec-core >= 2.5.3

### ISSUE THREAD

**Issue by DanBurton** [NorfairKing/validity#32]
It seems that `mapSpecTree` was removed from Test.Hspec.Core.Spec in hspec-core-2.5.3. (Which is odd, because this is a breaking change which should have warranted a major version bump.)

```
[ 2 of 14] Compiling Test.Validity.Utils ( src/Test/Validity/Utils.hs, dist/build/Test/Validity/Utils.o )

src/Test/Validity/Utils.hs:66:18: error:
    • Variable not in scope:
        mapSpecTree
          :: (SpecTree () -> SpecTree ()) -> SpecWith () -> SpecWith ()
    • Perhaps you meant ‘mapSpecItem’ (imported from Test.Hspec.Core.Spec)
   |
66 | failsBecause s = mapSpecTree go
   |                  ^^^^^^^^^^^
```

**Comment by NorfairKing** [comment]
Hi there! Thanks for the report! A PR would be welcome.

**Comment by NorfairKing** [comment]
@DanBurton Are you sure you want to put a bound on `hspec` instead of just fixing `genvalidity-hspec`?

**Comment by DanBurton** [comment]
@NorfairKing I would like to fix `genvalidity-hspec`; putting a bound on `hspec` is just temporary.

**Comment by NorfairKing** [comment]
@DanBurton What's the next step, now that #33 is merged?

**Comment by DanBurton** [comment]
Upload a new version of genvalidity-hspec to hackage, and once that is done I can go ahead and lift the constraint on hspec.

**Comment by NorfairKing** [comment]
There we go, 0.6.1.1 is on Hackage

**Comment by DanBurton** [comment]
Great, thanks!
